compiler/src/dmd/importc.d

```diff
@@ -538,6 +538,13 @@ Dsymbol handleSymbolRedeclarations(ref Scope sc, Dsymbol s, Dsymbol s2, ScopeDsy
     if (vd && vd.isCmacro())
         return vd2;
 
+    // function-like C macro? see https://issues.dlang.org/show_bug.cgi?id=24505
+    if (s.isCsymbol() && s.isTemplateDeclaration())
```

Comment:
This assumes all template declarations in C files must be function-like macros.

Reply:
And this almost certainly only handles the case where the macro comes lexically *after* the regular symbol.

Reply:
IIRC, macros are pushed to a special buffer, then processed last.